File: zs-leetcode-go/MinimumDepthofBinaryTree111/MinimumDepthofBinaryTree.go
package MinimumDepthofBinaryTree111

import "for-an-offer/zs-leetcode-go/types"

func MinimumDepthofBinaryTree(root *types.TreeNode) int {
	if root == nil {
		return 0
	}
	left := MinimumDepthofBinaryTree(root.Left)
	right := MinimumDepthofBinaryTree(root.Right)
	if left == 0 || right == 0 {
		return left + right + 1
	}
	if left < right {
		return left + 1
	}
	return right + 1
}

File: sjq-leetcode-C++/minimum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        if(root == NULL) return 0;
        if(root->left==NULL and root->right==NULL) return 1;
        if(root->left==NULL or root->right==NULL) return 1+max(minDepth(root->left), minDepth(root->right));
        return min(minDepth(root->left), minDepth(root->right))+1;
    }
};
